packages/turbo/src/BridgeComponent.ts

```diff
@@ -0,0 +1,89 @@
+import { Component } from 'react';
+import type { EmitterSubscription } from 'react-native';
+import type {
+  VisitableViewModule,
+  StradaEvent,
+  StradaMessages,
+  StradaReactComponentProps,
+} from './types';
+import { getNativeModule, registerMessageEventListener } from './common';
+
+const RNVisitableViewModule = getNativeModule<VisitableViewModule>(
+  'RNVisitableViewModule'
+);
+
+class BridgeComponent extends Component {
+  private name: string;
+  private url: string;
+  private sessionHandle: string;
+  private eventHandlerEventSubscription?: EmitterSubscription;
+  private previousMessages: StradaMessages = {};
+
+  constructor(props: StradaReactComponentProps) {
+    super(props);
+    this.url = props.url;
+    this.name = props.name;
+    this.sessionHandle = props.sessionHandle;
+    this.didReceive = this.didReceive.bind(this);
+    this.onReceive = this.onReceive.bind(this);
+    this.replyTo = this.replyTo.bind(this);
+  }
+
+  componentDidMount() {
+    this.eventHandlerEventSubscription = registerMessageEventListener(
+      this.name,
+      this.didReceive
+    );
+  }
+
+  componentWillUnmount() {
+    this.eventHandlerEventSubscription?.remove();
+  }
+
+  didReceive(e: object) {
+    const message = e as StradaEvent;
+    if (message.data.metadata.url !== this.url) {
+      return;
+    }
+    this.previousMessages[message.event] = message;
+    this.onReceive(message);
+  }
+
+  onReceive(_e: StradaEvent) {
+    throw new Error(
+      `BridgeComponent.onReceive must be overridden by ${this.name}`
+    );
+  }
+
+  replyTo(event: string, data?: object) {
```

Comment:
Strada native provides a few helpers to reply to messages. We have implemented `replyTo` here, but we could consider adding the rest to have parity with Strada native: https://github.com/hotwired/strada-android/blob/main/docs/BUILD-COMPONENTS.md?plain=1#L103-L112

```kotlin
replyTo("eventName")
replyTo("eventName", newData)
replyWith(originalMessage)
replyWith(originalMessage.replacing(data = newData))
replyWith(originalMessage.replacing(event = "newEventName"))
```

Reply:
We can implement this later though if we're pressed on time :)